ui/app/AppLayouts/Onboarding/views/ConfirmPasswordView.qml

```diff
@@ -0,0 +1,136 @@
+
+Page {
+    id: root
+
+    property string password
+
+    signal backClicked()
+
+    anchors.fill: parent
+
+    Component.onCompleted: confPswInput.forceActiveFocus(Qt.MouseFocusReason)
+
+    Column {
+        id: view
+        spacing: 4 * Style.current.padding
+        width: 416
+        anchors.centerIn: parent
+
+        StatusBaseText {
+            anchors.horizontalCenter: parent.horizontalCenter
+            text: qsTr("Have you written down your password?")
+            font.pixelSize: 22
+            font.bold: true
+            color: Theme.palette.directColor1
+        }
+
+        Column {
+            anchors.horizontalCenter: parent.horizontalCenter
+            spacing: Style.current.padding
+
+            StatusBaseText {
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: qsTr("You will never be able to recover your password if you loose it.")
+                font.pixelSize: 12
+                color: Theme.palette.dangerColor1
+            }
+
+            StatusBaseText {
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: qsTr("If you need to, write it using pen and paper and keep in a safe place.")
+                font.pixelSize: 12
+                color: Theme.palette.baseColor1
+            }
+
+            StatusBaseText {
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: qsTr("If you loose your password you will loose access to  your Status profile.")
+                font.pixelSize: 12
+                color: Theme.palette.baseColor1
+            }
+        }
+
+        // TODO replace with StatusInput as soon as it supports password
+        Input {
+            id: confPswInput
+            width: parent.width
+            enabled: !submitBtn.loading
+            placeholderText: submitBtn.loading ? qsTr("Connecting...") : qsTr("Confirm you password (again)")
+            textField.echoMode: TextInput.Password
+            textField.validator: RegExpValidator { regExp: /^[!-~]+$/ } // That incudes NOT extended ASCII printable characters less space
+            keepHeight: true
+            textField.rightPadding: showHideCurrentIcon.width + showHideCurrentIcon.anchors.rightMargin + Style.current.padding / 2
+
+            StatusFlatRoundButton {
+                id: showHideCurrentIcon
+                visible: confPswInput.text !== ""
+                anchors.verticalCenter: parent.verticalCenter
+                anchors.right: parent.right
+                anchors.rightMargin: 16
+                width: 24
+                height: 24
+                icon.name: confPswInput.textField.echoMode === TextInput.Password ? "show" : "hide"
+                icon.color: Theme.palette.baseColor1
+
+                onClicked: confPswInput.textField.echoMode === TextInput.Password ? confPswInput.textField.echoMode = TextInput.Normal : confPswInput.textField.echoMode = TextInput.Password
```

Comment:
I feel like we can simplify all the conditions here and above by creating a small property at the top of `confPswInput` called `showPassword`. It would default to `false` and then here you just have to do:
```qml
onClicked: confPswInput.showPassword = !confPswInput.showPassword
```
Also:
```qml
textField.echoMode: showPassword  ? TextInput.Normal : TextInput.Password
```
And other places where it can be simplified.

What do you think?

Reply:
I like your suggestion.. Cleaner!!